stan/math/prim/fun/inv_square.hpp

```diff
@@ -18,11 +18,27 @@ inline double inv_square(double x) { return inv(square(x)); }
  * @param x container
  * @return 1 / the square of each value in x.
  */
-template <typename T>
+template <typename T,
+          require_not_container_st<is_container, std::is_arithmetic, T>...>
 inline auto inv_square(const T& x) {
   return inv(square(x));
 }
 
+/**
+ * Version of inv_square() that accepts Eigen Matrix/Array objects or
+ * expressions.
+ *
+ * @tparam T Type of x
+ * @param x Eigen Matrix/Array or expression
+ * @return 1 / the square of each value in x.
+ */
+template <typename T,
+          require_container_st<is_container, std::is_arithmetic, T>...>
+inline auto inv_square(const T& x) {
+  return apply_vector_unary<T>::apply(
+      x, [&](const auto& v) { return v.derived().array().square().inverse(); });
+}
+
```

Comment:
This one is not needed as both `inv` and `square` use `apply_vector_unary`. So the simple implementation of `inv_square` already works for all input types.

Reply:
Good call, will revert

Reply:
Actually, I feel like this would be more efficient. Since with ```inv(square(x))```, wouldn't the matrix ```x``` would get traversed twice? Once for ```square``` and once for ```inv```?

Reply:
Yes, but just until we remove all those `eval()`s. I am leaving it up to you if you want to leave it in.

Reply:
That's the case for Eigen objects, but not for ```std::vector<double>``` or ```std::vectors``` of Eigen objects, since they'll get evaluated and returned between each call